font-types/src/path.rs

```diff
@@ -0,0 +1,24 @@
+/// Interface for accepting a sequence of path commands.
```

Comment:
This trait looks to me a lot like the "pen" protocol in fonttools and font-related python libraries. I wonder where the "sink" metaphor comes from? I'm sure both have their own "traditions". FWIW the "pen" is familiar to type designers and font engineers. Not immediately suggesting that you rename, just curious.

Reply:
The metaphor comes from the concept of sources and sinks-- the former being something that generates output and the latter something that receives input. If the name is not appropriately descriptive, a reasonable alternative in Rust land would be to call it `PathBuilder` but I usually like to reserve that for concrete types.

Btw, just checked out the pen module in fonttools. That's pretty nifty! Composable path adapters are a great idea and much easier to implement efficiently now that we have GATs.

Reply:
https://en.wikipedia.org/wiki/Sink_(computing). To me this is more commonly associated with data pipelines and while not _wrong_ here I'm not sure it's a perfect fit either.

I don't think `trait PathBuilder` is all that unreasonable? - being a trait is a very clear message about the concrete'ness of it.

`trait Pen` is also entirely reasonable IMHO and the consistency with prior art is a non-trivial plus.

Reply:
Thinking on this a bit, Pen reads nicely to me. I’ll make the change.